api/api.go

```diff
@@ -0,0 +1,65 @@
+package api
```

Comment:
There are lots of changes in this PR that are not related to the Shutdown function?

Reply:
They are layered branches in which it contains all the commits from the previous PRs as well.

Reply:
Thats why as a reviewer I dont think we should be looking at these pull requests until the previous prs are reviewed, updated, approved and merged

Reply:
Or change the base branch to the branch that this branch was created off - that way the diff will be between the two branches which actually care about at this point in the git history

Reply:
Changing the base branch did the trick. Thank you